### PR TITLE
[19780] Add interface whitelist by name

### PR DESCRIPTION
**Pull request by elianalf**
The feature gives users the possibility to add network interfaces to the whitelist by interface name as well, not just by their IP address.

Merge after:
- https://github.com/eProsima/Fast-DDS/pull/3973

### CODE REVIEW

File: docs/fastdds/transport/whitelist.rst [hunk 1]
```diff
@@ -36,8 +39,32 @@ For example:
...
+    .. literalinclude:: /../code/XMLTester.xml
+      :language: xml
+      :start-after: <!-->WHITELIST-NAME
+      :end-before: <!--><-->
+      :lines: 2-3,5-
+      :append: </profiles>
+
+.. important::
+
+  The values in this list should match the IP addresses or interface names on your machine for the corresponding networks. If none of the values in the whitelist match the interfaces on your machine, all the interfaces in the whitelist are filtered out and an invalid interface is assigned.
```
Comment (juanlofer-eprosima) [inline docs/fastdds/transport/whitelist.rst:65]:
```suggestion
  If none of the values in the transport descriptor's whitelist match the interfaces on your machine, all the interfaces in the whitelist are filtered out and no communication will be established through that transport.
```

Comment (juanlofer-eprosima) [reply docs/fastdds/transport/whitelist.rst:65]: Just to shorten it a bit, and saying we add an invalid ip is perhaps too low level for a user.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

**Review by juanlofer-eprosima — DISMISSED** [review]
LGTM

File: docs/fastdds/transport/whitelist.rst [hunk 2]
```diff
@@ -9,14 +9,17 @@ Interface Whitelist
...
-This is achieved by adding the interfaces' IP addresses to the |SocketTransportDescriptor::interfaceWhiteList-api|
+This is achieved by adding the interfaces to the |SocketTransportDescriptor::interfaceWhiteList-api|
 field in the :ref:`transport_tcp_transportDescriptor` or :ref:`transport_udp_transportDescriptor`.
 Thus, the communication interfaces used by the |DomainParticipants| whose |TransportDescriptorInterface-api| defines an
-|SocketTransportDescriptor::interfaceWhiteList-api| is limited to the interfaces' IP addresses defined in that list,
+|SocketTransportDescriptor::interfaceWhiteList-api| is limited to the interfaces' addresses defined in that list,
 therefore avoiding the use of the rest of the network interfaces available in the system.
-The values on this list should match the IPs of your machine in that networks.
+It is possible to add the interfaces to |SocketTransportDescriptor::interfaceWhiteList-api| specifying the IP addresses
+or the interface names.
```
Comment (EduPonz) [inline docs/fastdds/transport/whitelist.rst:18]:
```suggestion
The interfaces in |SocketTransportDescriptor::interfaceWhiteList-api| can be specified both by IP address
or interface name.
```

File: docs/fastdds/transport/whitelist.rst [hunk 3]
```diff
@@ -9,14 +9,17 @@ Interface Whitelist
...
 For example:
 
+* Interface whitelist filled with IP address:
+
```
Comment (EduPonz) [inline docs/fastdds/transport/whitelist.rst:22]: Indent the tabs so they are nested withing the bullet list

File: docs/fastdds/transport/whitelist.rst [hunk 4]
```diff
@@ -36,6 +39,32 @@ For example:
       :lines: 2-3,5-
       :append: </profiles>
 
+* Interface whitelist filled with interface names:
+
+.. tabs::
```
Comment (EduPonz) [inline docs/fastdds/transport/whitelist.rst:44]: Same as before, indent the tabs

File: docs/fastdds/transport/whitelist.rst [hunk 5]
```diff
@@ -36,6 +39,32 @@ For example:
...
+      :language: xml
+      :start-after: <!-->WHITELIST-NAME
+      :end-before: <!--><-->
+      :lines: 2-3,5-
+      :append: </profiles>
+
+.. important::
+
+  If none of the values in the transport descriptor's whitelist match the interfaces on your machine,
+  all the interfaces in the whitelist are filtered out and no communication will be established through that transport.
```
Comment (EduPonz) [inline docs/fastdds/transport/whitelist.rst:66]:
```suggestion
  If none of the values in the transport descriptor's whitelist matches the interfaces present on the host,
  then all the interfaces in the whitelist are filtered out and therefore no communication will be established through that transport.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
A couple of NITs

**Review by EduPonz — DISMISSED** [review]
LGTM with green CI

**Review by EduPonz — APPROVED** [review]
LGTM with green CI

**Comment by EduPonz** [comment]
@mergifyio rebase master

**Pull request merged**